test/foundry/strategies/QVImpactStream.t.sol

```diff
@@ -0,0 +1,231 @@
+        vm.mockCall(
+            mockAlloAddress, abi.encodeWithSelector(IAllo.isPoolManager.selector, 1, poolManager), abi.encode(true)
+        );
+        _;
+    }
+
+    function test_BatchAddAllocatorWhenCalledByPoolManager() external callWithPoolManager {
+        // if AllocatorAdded event emits with the correct parameters then _addAllocator was also called
+        // with the correct parameters
+        vm.expectEmit(true, true, true, true);
+        emit AllocatorAdded(recipient1, poolManager);
+
+        address[] memory _recipients = new address[](1);
+        _recipients[0] = recipient1;
+
+        vm.prank(poolManager);
+        qvImpactStream.batchAddAllocator(_recipients);
+
+        assertTrue(qvImpactStream.allowedAllocators(recipient1));
+    }
+
+    function test_BatchRemoveAllocatorWhenCalledByPoolManager() external callWithPoolManager {
+        // if AllocatorRemoved event emits with the correct parameters then _removeAllocator was also called
+        // with the correct parameters
+        vm.expectEmit(true, true, true, true);
+        emit AllocatorRemoved(recipient1, poolManager);
+
+        address[] memory _recipients = new address[](1);
+        _recipients[0] = recipient1;
+
+        vm.prank(poolManager);
+        qvImpactStream.batchRemoveAllocator(_recipients);
+
+        assertFalse(qvImpactStream.allowedAllocators(recipient1));
+    }
+
+    function test_SetPayoutsRevertWhen_PayoutSetIsTrue() external callWithPoolManager {
+        stdstore.target(address(qvImpactStream)).sig("payoutSet()").checked_write(true);
+        vm.expectRevert(QVImpactStream.PAYOUT_ALREADY_SET.selector);
+
+        /// make it after allocation finished
+        vm.warp(block.timestamp + allocationWindow + 1 days);
+
+        vm.prank(poolManager);
+        qvImpactStream.setPayouts(payouts);
+    }
+
+    function test_SetPayoutsRevertWhen_PayoutAmountIsZero() external callWithPoolManager {
+        vm.expectRevert(abi.encodeWithSelector(Errors.RECIPIENT_ERROR.selector, recipient1));
+        payouts.push(QVImpactStream.Payout({recipientId: recipient1, amount: 0}));
+
+        /// make it after allocation finished
+        vm.warp(block.timestamp + allocationWindow + 1 days);
+
+        vm.prank(poolManager);
+        qvImpactStream.setPayouts(payouts);
+    }
+
+    function test_SetPayoutsRevertWhen_RecipientStatusIsNotAccepted() external callWithPoolManager {
+        vm.expectRevert(abi.encodeWithSelector(Errors.RECIPIENT_ERROR.selector, recipient1));
+
+        /// make it after allocation finished
+        vm.warp(block.timestamp + allocationWindow + 1 days);
+
+        /// since the recipient is not registered, his status should be NONE
+        /// so setting payouts should fail
+        vm.prank(poolManager);
+        qvImpactStream.setPayouts(payouts);
+    }
+
+    function test_SetPayoutsRevertWhen_TotalPayoutIsGreaterThanPoolAmount() external {
+        // it should revert
+        vm.skip(true);
```

Comment:
Why those tests skipped?

Reply:
Not really sure what would be the best way to mock `_getRecipientStatus` ?